torch/lib/THPP/tensors/generic/THCTensor.cpp

```diff
@@ -30,6 +30,12 @@ auto THCTensor<real>::clone_shallow() -> THCTensor* {
 }
 
 template<>
+auto THCTensor<real>::contiguous() const -> std::unique_ptr<Tensor> {
+  return std::unique_ptr<Tensor>(
+      new THCTensor(state, THCTensor_(newContiguous)(state, tensor)));
```

Comment:
Yeah. I don't see an easy way to avoid this. We have the same issue with clone() and clone_shallow()